### PR TITLE
Fix key padding mask doc

### LINKED ISSUE #141018: The description of key_dedding_mask in float mode is incorrect
### 📚 The doc issue

The problematic [document](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html#torch.nn.MultiheadAttention) is about the description of key_padding_mask.
key_padding_mask：For a float mask, it will be directly added to the corresponding key value.

issue：I tested this float type and the results showed that it was not directly added to the key value, but did change the weight value in some form. The code and results are shown below.
```
import torch
import torch.nn as nn
import torch.nn.functional as F

multiheadattention=nn.MultiheadAttention(embed_dim=2,
num_heads=1,bias=False,batch_first=True)
data=torch.tensor([[[1,1.],[2,0.]]])
multiheadattention.in_proj_weight.data=torch.tensor(
[[1.,2],[0,1],[1,0],[1,1],[2,1],[2,0]])
multiheadattention.out_proj.weight.data=torch.tensor([[1,2],
[3,4.]])
keypaddingmask=torch.tensor([[1,2.]])
out1=multiheadattention(data,data,data,
need_weights=True,key_padding_mask=keypaddingmask)#When removing key_padding_mask, the returned weight tensor([[[0.1070, 0.8930], [0.1956, 0.8044]]] is correct, but adding key_padding_mask will result in an incorrect weight tensor([[[0.0422, 0.9578], [0.0821, 0.9179]]] as described in the document.
print(out1)
```

image

### Suggest a potential alternative/fix

Update the description about key_padding_mask.

### PR DESCRIPTION
**Pull request by joldov**
Fixes #141018 
- Updated the docstring to correctly state that float masks modify attention weights instead of key values, with examples of typical values (e.g., -inf for masking)
- Verified and ensured clarity in how boolean masks affect sequence positions
- Rebuilt the documentation locally to confirm updates render properly

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

File: torch/ao/nn/quantizable/modules/activation.py [hunk 1]
```diff
@@ -309,7 +309,9 @@ def forward(
               the embedding dimension. :math:`(N, S, E)` if ``batch_first`` is ``True``.
             - key_padding_mask: :math:`(N, S)` where N is the batch size, S is the source sequence length.
               If a BoolTensor is provided, the positions with the
-              value of ``True`` will be ignored while the position with the value of ``False`` will be unchanged.
+              value of ``True`` will be ignored (masked out) while the position with the value of ``False`` will be unchanged.
```
Comment (cpuhrsch) [inline torch/ao/nn/quantizable/modules/activation.py:312]: I think you meant to apply this to the MHA in torch/nn/modules/activation.py ?
